src/core/metrics.cc

```diff
@@ -196,6 +226,49 @@ Metrics::SetMetricsInterval(uint64_t metrics_interval_ms)
   singleton->metrics_interval_ms_ = metrics_interval_ms;
 }
 
+bool
+Metrics::InitializeCacheMetrics(std::shared_ptr<RequestResponseCache> response_cache)
+{
+  if (response_cache == nullptr) {
+     LOG_WARNING << "error initializing cache metrics, cache metrics will not be "
+                << "available: cache was nullptr";
+    return false;  
+  }
+
+  // TODO: Populate labels
+  const std::map<std::string, std::string> cache_labels;
+  cache_num_entries_global_ = &cache_num_entries_family_.Add(cache_labels);
+  cache_thread_exit_.store(false);
+
+  // Start a separate thread for updating cache metrics at specified interval
+  cache_thread_.reset(new std::thread([this, response_cache] {
```

Comment:
Since we already has a running thread for pulling DCGM metrics and both threads share the same iteration interval, would that make sense to have only one polling thread and perform both DCGM polling and cache polling (if enable)?

Reply:
I think that would make sense as long as the thread doesn't eventually become bogged down with too much work to complete within the interval

Reply:
@GuanLuo a lot of the DCGM logic is surrounded by `TRITON_ENABLE_GPU_METRICS` ifdefs, makingit a little more tedious to separate cache metrics which can be enabled with CPU only. Do you think we should still do so?

Reply:
I agree that it would be nice to only have a single metrics thread that was handing both gpu and cache polling.

Reply:
OK, will do

Reply:
OK, will do. Thanks for the feedback Guan and David 🙂

Reply:
What I have in mind is to extract the thread logic from both `EnableXXX()` to a separate function. Inside the function, it checks for the shared flags on whether cache / GPU metrics are enabled and perform corresponding pollings. Then the `EnableXXX` functions will set the flag and create the thread if it hasn't been created